misk/src/main/kotlin/misk/web/HttpCall.kt

```diff
@@ -11,13 +11,21 @@ import okhttp3.RequestBody
 import okio.BufferedSink
 import okio.BufferedSource
 
+/** The type of socket on which a request arrived.
+ */
+enum class SourceSocketType {
+  NETWORK,
+  UNIX
+}
+
 /**
  * A live HTTP call from a client for use by a chain of network interceptors.
  */
 interface HttpCall {
 
   /** Immutable information about the incoming HTTP request. */
   val url: HttpUrl
+  val sourceSocketType: SourceSocketType
```

Comment:
Since you'll be able to extract it, you may want to include the listener name using something like:
```
data class Listener(socketType: SourceSocketType, name String)

interface HttpCall {
  // ...
  val listener: Listener
  // ...
}
```

This would let the interceptor trust only a specific listener, not any unix socke tone.